src/org/exist/client/InteractiveClient.java

```diff
@@ -2213,23 +2086,22 @@ private void connectToDatabase() {
         try {
             connect();
         } catch (final Exception cnf) {
-            if (startGUI && frame != null) {
+            if (options.startGUI && frame != null) {
                 frame.setStatus("Connection to database failed; message: " + cnf.getMessage());
             } else {
                 System.err.println("Connection to database failed; message: " + cnf.getMessage());
             }
             cnf.printStackTrace();
-            System.exit(1);
+            System.exit(SystemExitCodes.CATCH_ALL_GENERAL_ERROR_EXIT_CODE);
         }
     }
 
     /**
      * Main processing method for the InteractiveClient object
      *
-     * @param args arguments from main()
      * @return true on success, false on failure
      */
-    public boolean run(final String args[]) throws Exception {
+    public boolean run() throws Exception {
```

Comment:
@shabanovd Sorry, this a little off topic from the issue at hand, but I'm just curious: 

The rules you mention, where can I find these?

You can imagine I'm seeing this as an opportunity for discussion, so I'd be grateful to consult whatever documentation there already is in this area.